### PR TITLE
improve locale utility to fix #1691 and #1694

### LINKED ISSUE #1691: [BUG][Ruby Client]Failed to get localized translation when requesting locale(JA, FR...) with upper letters by getString() and translate() interface.
**Describe the bug**
Failed to get localized translation when requesting locale(JA, FR...) with upper letters by getString() and translate() interface.

**To Reproduce**
1. Load config file.
Singleton.load_config(<config file>, <mode>)
2. The translation of en/ja locale:
```
en: about.description: "Use this area to provide additional information"
ja:  "about.description" : "この領域を使用して追加情報を提供する"
fr:  "about.description" : "Utilisez cette zone pour fournir des informations supplémentaires"
```
3. Invoke below interface:
```
expect(SgtnClient::Translation.getString("about", "about.message", "JA")).to eq("この領域を使用して追加情報を提供する")
expect(Singleton.t("about.message", "about", "FR")).to eq("there is one cat in the room")
```
4. See error:
```
expected: "この領域を使用して追加情報を提供する"
            got: "Use this area to provide additional information"

expected: "Utilisez cette zone pour fournir des informations supplémentaires"
            got: "Use this area to provide additional information"
```

**Expected behavior**
expect(SgtnClient::Translation.getString("about", "about.message", "JA")).to eq("この領域を使用して追加情報を提供する")
expect(Singleton.t("about.message", "about", "FR")).to eq("there is one cat in the room")

### PR DESCRIPTION
**Pull request by Xiaochao8**

### CODE REVIEW

File: lib/sgtn-client/core/config.rb [hunk 1]
```diff
@@ -182,6 +183,8 @@ def available_locales
             def bundles.locales
               @locales ||= reduce(Set.new) { |locales, id| locales << id.locale }
             end
+            changed
+            notify_observers(self, :available_locales)
```
Comment (jessiejuachon) [inline lib/sgtn-client/core/config.rb:187]:
```suggestion
            notify_observers(:available_locales)
```

File: lib/sgtn-client/util/locale-util.rb [hunk 2]
```diff
@@ -47,6 +53,18 @@ def self.get_fallback_locale
...
     end
 
-    private_class_method :get_best_match
+    def self.lowercase_locales_map
+      @lowercase_locales_map ||= Config.available_locales.each_with_object({}) do |locale, memo|
+        memo[locale.to_s.downcase] = locale
+      end
+    end
+
+    def self.reset_lowercase_locales_map(_config, type)
```
Comment (jessiejuachon) [inline lib/sgtn-client/util/locale-util.rb:62]:
```suggestion
    def self.reset_lowercase_locales_map(type)
```

**Review by jessiejuachon — CHANGES_REQUESTED** [review]

File: lib/sgtn-client/core/config.rb (hunk 1 above)
Comment (Xiaochao8) [reply lib/sgtn-client/core/config.rb]: fixed.

File: lib/sgtn-client/util/locale-util.rb (hunk 2 above)
Comment (Xiaochao8) [reply lib/sgtn-client/util/locale-util.rb]: fxied

**Review by jessiejuachon — APPROVED** [review]

**Review by deminyan — APPROVED** [review]

**Pull request merged**